src/layouts/hass-tabs-subpage-data-table.ts

```diff
@@ -333,10 +333,10 @@ export class HaTabsSubpageDataTable extends KeyboardShortcutMixin(LitElement) {
                 : nothing
             )}
             <ha-md-menu-item
-              .value=${undefined}
+              .value=${""}
               .clickAction=${this._handleGroupBy}
-              .selected=${this._groupColumn === undefined}
-              class=${classMap({ selected: this._groupColumn === undefined })}
+              .selected=${!this._groupColumn}
```

Comment:
below the collapse items where previously disabled when there was no grouping, we should probably update those too?